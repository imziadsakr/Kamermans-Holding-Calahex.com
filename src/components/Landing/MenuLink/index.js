

import React, { useMemo } from 'react' ;

import { Link } from 'react-router-dom';

import { 
    Grid ,
    Box, 
    useMediaQuery
}
from '@mui/material';

import { makeStyles } from '@mui/styles';


const useStyles = makeStyles((theme) => ({
    root : {
        marginTop : "30px" ,

        marginLeft : "10%",
        marginRight : "10%",
        textAlign : "center" ,
        fontSize : "20px" ,
        color : theme.palette.primary.main ,
        fontFamily : "Roboto,  Helvetica, Arial, sans-serif" ,

        "& .MuiGrid-item" : {
            paddingTop : "15px",
        },
        "& a" : {
            textDecoration : "none" ,
            color : theme.palette.primary.main
        }
    } , 
    menuLink : {
        textAlign : "center !important",
    } ,
}));


const MenuLink = () => {

    const classes = useStyles() ;

    const match1 = useMediaQuery("(min-width : 840px)");
    const match2 = useMediaQuery("(min-width : 400px)");

    const menuLinkList = [
        {
            linkName : "Crpyto Exchange",
            link : "/cryptoexchange"
        } ,
        {
            linkName : "Wallet",
            link : "/wallet"
        } ,
        {
            linkName : "Buy Crpyto",
            link : "/buycrypto"
        },
        {
            linkName : "CAX Rewards",
            link : "/caxrewards"
        }
    ]

    return (
        <Box  className={classes.root}>
            <Grid container>
                {
                    menuLinkList.map(( element , index ) => {
                        return (
                            <Grid item xs={match1 ? 3 : match2 ? 6 : 12 } className={classes.menuLink} key={index} >
                                <Link to={element.link}>
                                    { element.linkName }
                                </Link>
                            </Grid>
                        )
                    })
                }
            </Grid>
        </Box>
    )
}


export default MenuLink ;

