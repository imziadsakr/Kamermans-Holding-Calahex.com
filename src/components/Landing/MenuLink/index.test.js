import React from 'react';
import renderer from 'react-test-renderer';

import theme from '../../../utils/theme';
import { ThemeProvider } from '@mui/material';

import { BrowserRouter } from 'react-router-dom';

import MenuLink from '.' ;

describe("MenuLink", () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <ThemeProvider theme={theme}>
                        <MenuLink />
                    </ThemeProvider>
                </BrowserRouter>    
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})