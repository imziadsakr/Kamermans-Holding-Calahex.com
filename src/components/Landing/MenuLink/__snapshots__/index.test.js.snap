// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLink renders correctly 1`] = `
<div
  className="makeStyles-root-1 MuiBox-root css-0"
>
  <div
    className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-menuLink-2 css-1idn90j-MuiGrid-root"
    >
      <a
        href="/cryptoexchange"
        onClick={[Function]}
      >
        Crpyto Exchange
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-menuLink-2 css-1idn90j-MuiGrid-root"
    >
      <a
        href="/wallet"
        onClick={[Function]}
      >
        Wallet
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-menuLink-2 css-1idn90j-MuiGrid-root"
    >
      <a
        href="/buycrypto"
        onClick={[Function]}
      >
        Buy Crpyto
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-menuLink-2 css-1idn90j-MuiGrid-root"
    >
      <a
        href="/caxrewards"
        onClick={[Function]}
      >
        CAX Rewards
      </a>
    </div>
  </div>
</div>
`;
